crates/matrix-sdk-crypto/src/olm/session.rs

```diff
@@ -285,3 +275,70 @@ pub struct PickledSession {
     /// The Unix timestamp when the session was last used.
     pub last_use_time: SecondsSinceUnixEpoch,
 }
+
+#[cfg(test)]
+mod tests {
+    use matrix_sdk_test::async_test;
+
+    use ruma::{device_id, user_id};
+    use serde_json::{self, Value};
+    use vodozemac::olm::{OlmMessage, SessionConfig};
+
+    use crate::{
+        identities::ReadOnlyDevice, olm::Account,
+        types::events::room::encrypted::ToDeviceEncryptedEventContent,
+    };
+
+    #[async_test]
+    async fn test_encryption_and_decryption() {
+        use ruma::events::dummy::ToDeviceDummyEventContent;
+
+        let alice =
+            Account::with_device_id(user_id!("@alice:localhost"), device_id!("ALICEDEVICE"));
+        let mut bob = Account::with_device_id(user_id!("@bob:localhost"), device_id!("BOBDEVICE"));
+
+        bob.generate_one_time_keys(1);
+        let one_time_key = *bob.one_time_keys().values().next().unwrap();
+        let sender_key = bob.identity_keys().curve25519;
+        let mut alice_session = alice.create_outbound_session_helper(
+            SessionConfig::default(),
+            sender_key,
+            one_time_key,
+            false,
+            alice.device_keys(),
+        );
+
+        let alice_device = ReadOnlyDevice::from_account(&alice);
+
+        let message = alice_session
+            .encrypt(&alice_device, "m.dummy", ToDeviceDummyEventContent::new(), None)
+            .await
+            .unwrap()
+            .deserialize()
+            .unwrap();
+
+        let content = if let ToDeviceEncryptedEventContent::OlmV1Curve25519AesSha2(c) = message {
```

Comment:
This would be slightly easier to read with the `let ... else` pattern, which was recently added https://doc.rust-lang.org/beta/rust-by-example/flow_control/let_else.html